addons/payment_stripe/models/payment_transaction.py

```diff
@@ -200,7 +200,7 @@ def _stripe_create_payment_intent(self):
             'payment_intents',
             payload=self._stripe_prepare_payment_intent_payload(),
             offline=self.operation == 'offline',
-            idempotency_key=payment_utils.generate_idempotency_key(self, scope='payment_intents'),
+            idempotency_key=payment_utils.generate_idempotency_key(self, scope='payment_intents', payment_method=self.token_id.stripe_payment_method),
```

Comment:
We don't want to keep users from retrying a payment. In this case, the payment method differs but there could be other possibilities (like a customer that does not have enough money in his bank account. He will get an "Insufficient funds" error, then make an instant transfer and retry. It won't work since the payment method is the same).

The idempotency key prevents issues where the customer is charged twice for the same thing. In this case, we don't want to prevent anything since the customer is on the page. I discussed this matter with the payment team and we suggest using idempotency keys only for offline payment when the customer is not in front of the payment page (e.g. when the cron charges the customer for his subscription).
```suggestion
            idempotency_key=payment_utils.generate_idempotency_key(
                self, scope='payment_intents_token'
            ) if self.operation == 'offline' else None,
```